security/pkg/nodeagent/model/secret.go

```diff
@@ -24,6 +24,9 @@ type SecretItem struct {
 
 	RootCert []byte
 
+	// Certificate Revocation List used by envoy when the RootCert is provided
```

Comment:
`rootcert` is provided, you mean the crl needs to be signed by the same root cert issuing certificate? if so, can you make this comment more explicit?

do we have validation . check to enforce that and signal error if not?

Reply:
The comment is referring to the fact that when a RootCert we aren't doing mtls, so any CRL provided would not be used.  Other than that it's not describing any restrictions on what certificate signed the CRL.  I will make the comment more clear.

From looking at the Envoy code I don't see where such a check is made against their `certificate_revocation_list_`.  I would prefer to keep as is.